smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/integration/DocumentMemberSerVisitor.java

```diff
@@ -216,7 +216,11 @@ public final String memberShape(MemberShape shape) {
     public String timestampShape(TimestampShape shape) {
         HttpBindingIndex httpIndex = HttpBindingIndex.of(context.getModel());
         Format format = httpIndex.determineTimestampFormat(shape, Location.DOCUMENT, defaultTimestampFormat);
-        return HttpProtocolGeneratorUtils.getTimestampInputParam(context, dataSource, shape, format);
+        String base = HttpProtocolGeneratorUtils.getTimestampInputParam(context, dataSource, shape, format);
+        if (format.equals(Format.EPOCH_SECONDS)) {
+            base = base + ".toString()";
+        }
```

Comment:
I wonder if this logic should be in [HttpProtocolGeneratorUtils.getTimestampInputParam](https://github.com/awslabs/smithy-typescript/blob/main/smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/integration/HttpProtocolGeneratorUtils.java#L74)

Reply:
Specifically, it seems like within that function, both DATE_TIME and HTTP_DATE return what would be a string. Unless there's somewhere where EPOCH_SECONDS needs to be a number?

Reply:
That makes sense. That will also allow for dropping the `.toString()` that's added [here](https://github.com/awslabs/smithy-typescript/blob/c874a2a0d60a9496ce9b59915f9ed4460e76b5b1/smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/integration/HttpBindingProtocolGenerator.java#L1417) which would clean up some extra `.toString()` calls that aren't really needed.